### PR TITLE
introduce host_device for pass

### PR DESCRIPTION
**Pull request by guotuofeng**
## Describe your changes
Introduce host device concept for pass so that each pass could choose override it or choose it.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: olive/passes/pytorch/sparsegpt.py [hunk 1]
```diff
@@ -70,9 +70,9 @@ def _default_config(cls, accelerator_spec: AcceleratorSpec) -> Dict[str, PassCon
                 description="Only prune layers whose name contains the given string(s).",
             ),
             # this is not the same as accelerator_spec.device which is the target device for inference
-            # compute_device is the device we want to run the algorithm on, does not affect the final model
-            # so accelerator_spec.device can be cpu but compute_device can be cuda for faster pass execution
-            "compute_device": PassConfigParam(
+            # device is the device we want to run the algorithm on, does not affect the final model
+            # so accelerator_spec.device can be cpu but device can be cuda for faster pass execution
+            "device": PassConfigParam(
```
Comment (trajepl) [inline olive/passes/pytorch/sparsegpt.py:75]: I forget what this device is? target_deivce(accelerators.device) or host_device? Or neither are? Could you remind me?

Looks like even we have target device and host device, we still cannot remove the `device` arguments for some particular passes.

**Comment by trajepl** [comment]
Also could you help double confirm:

With this PR, most of the `device` in passes configs should be unified somehow? Remove or use the same name?

**Comment by guotuofeng** [comment]
> Also could you help double confirm:
> 
> With this PR, most of the `device` in passes configs should be unified somehow? Remove or use the same name?

With this PR, each pass should have freedom to choose use "device" or anything else(for e.g. "target_device" for snpe). There is no strong requirement to unify them.

File: olive/passes/pytorch/sparsegpt.py (hunk 1 above)
Comment (guotuofeng) [reply olive/passes/pytorch/sparsegpt.py:75]: the compute_device should be host_device. I will let @jambayk to double confirm.

**Review by trajepl — APPROVED** [review]

**Pull request merged**